### PR TITLE
Added Behat integration

### PR DESCRIPTION
**Pull request by wouterj**
This PR adds the necessary classes to integrate with Behat, similar to the existing PHPunit integration. I hope this is something you want to add, it would be of great help for our test suite :)

If you want this, do you want me to add a Behat test set-up to functionally test the integration?

### CODE REVIEW

**Review by dmaicher — COMMENTED** [review]
Thanks for proposing this @wouterj :+1: 

Having this covered by a functional test would be nice. Will this require a lot of code?

File: README.md [hunk 1]
```diff
@@ -59,11 +61,24 @@ It also includes a `StaticArrayCache` that will be automatically configured as m
...
+2. Make sure you also have `phpunit/phpunit` available as a `dev` dependency (**versions 7, 8 and 9 are supported with the built in listener/extension**) to run your tests. 
    Alternatively this bundle is also compatible with `symfony/phpunit-bridge` and its `simple-phpunit` script. 
    (Note: you may need to make sure the phpunit-bridge requires the correct PHPUnit 7+ Version using the environment variable `SYMFONY_PHPUNIT_VERSION`). 
 
-5. That's it! From now on whatever changes you do to the database within each single testcase (be it a `WebTestCase` or a `KernelTestCase` or any custom test) are automatically rolled back for you :blush:
+3. That's it! From now on whatever changes you do to the database within each single testcase (be it a `WebTestCase` or a `KernelTestCase` or any custom test) are automatically rolled back for you :blush:
+
+#### Using the Bundle with Behat
+
+Enable the extension in your Behat config (e.g. `behat.yml`)
```
[the remark this answers is not in the document]
Comment (wouterj) [reply README.md:72]: Good points! I'll update it.

(and oh, I completely missed that part of the README, guess I should read the complete README next time :see_no_evil: )

File: composer.json [hunk 2]
```diff
@@ -17,11 +17,15 @@
...
   },
   "require-dev": {
+    "behat/behat": "^3.0",
     "phpunit/phpunit": "^7.0 || ^8.0 || ^9.0",
     "symfony/yaml": "^3.4 || ^4.3 || ^5.0",
     "symfony/phpunit-bridge": "^4.3 || ^5.0",
     "phpstan/phpstan": "^0.12"
   },
+  "conflict": {
+    "behat/behat": "<3.0"
```
[the remark this answers is not in the document]
Comment (wouterj) [reply composer.json:27]: Yes, indeed. I'll remove it (Behat 3 is out for so long that I think everyone actively working on their test suite is already on it).

**Comment by wouterj** [comment]
> Having this covered by a functional test would be nice. Will this require a lot of code?

I'll have a look, I think it'll have the same or less code as the PHPunit functional test.

Thanks for your review, I'll try to find some time this week to work on this PR.

**Comment by wouterj** [comment]
Updated the PR, I've added functional behat tests (similar to those existing in PHPunit). I couldn't understand what `PhpunitTest::testRollBackChangesWithReOpenedConnection` was testing, so I haven't copied this one to the Behat tests. I also didn't move the `testPreviousChangesAreRolledBack` tests to Behat, it's a bit non-behat-ish to do so. As all scenarios start with `Given there are 0 rows`, we already test it enough imho.

**Review by dmaicher — APPROVED** [review]
Nice :+1: Thanks

**Comment by dmaicher** [comment]
@mnapoli @dkarlovi what do you think about this? Any chance you could review the behat part? Since you commented [here](https://github.com/dmaicher/doctrine-test-bundle/issues/48) I guess you are using this bundle with behat? :blush:

**Comment by dmaicher** [comment]
@wouterj I added some changes on master (php8 support for example). Could you solve the conflicts please? I would like to merge this as well for release 6.4.0

**Review by dmaicher — APPROVED** [review]
Thanks @wouterj :+1:

**Comment by wouterj** [comment]
Thank you for your time reviewing & merging this PR @dmaicher!

**Pull request merged**